maat-scheduled-tasks/src/main/java/uk/gov/justice/laa/maat/scheduled/tasks/service/ApplicantBillingService.java

```diff
@@ -1,85 +1,74 @@
 package uk.gov.justice.laa.maat.scheduled.tasks.service;
 
-import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
-import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
-import org.springframework.transaction.annotation.Transactional;
 import uk.gov.justice.laa.maat.scheduled.tasks.client.CrownCourtLitigatorFeesApiClient;
-import uk.gov.justice.laa.maat.scheduled.tasks.dto.ApplicantBillingDTO;
-import uk.gov.justice.laa.maat.scheduled.tasks.dto.ResetApplicantBillingDTO;
 import uk.gov.justice.laa.maat.scheduled.tasks.entity.ApplicantBillingEntity;
 import uk.gov.justice.laa.maat.scheduled.tasks.enums.BillingDataFeedRecordType;
 import uk.gov.justice.laa.maat.scheduled.tasks.mapper.ApplicantMapper;
 import uk.gov.justice.laa.maat.scheduled.tasks.repository.ApplicantBillingRepository;
 
 import java.util.List;
 import uk.gov.justice.laa.maat.scheduled.tasks.request.UpdateApplicantsRequest;
-import uk.gov.justice.laa.maat.scheduled.tasks.utils.ResponseUtils;
 
 @Slf4j
 @Service
-@RequiredArgsConstructor
-public class ApplicantBillingService {
+public class ApplicantBillingService extends BillingService {
 
     private final ApplicantBillingRepository applicantBillingRepository;
-    private final BillingDataFeedLogService billingDataFeedLogService;
-    private final CrownCourtLitigatorFeesApiClient crownCourtLitigatorFeesApiClient;
     private final ApplicantMapper applicantMapper;
-    private static final String SENT_TO_CCLF_FAILURE_FLAG = "Y";
     private static final String REQUEST_LABEL = "applicant";
 
-    @Transactional
-    public void sendApplicantsToBilling(String userModified) {
-        List<ApplicantBillingDTO> applicants = findAllApplicantsForBilling();
+    public ApplicantBillingService(BillingDataFeedLogService billingDataFeedLogService,
+        CrownCourtLitigatorFeesApiClient crownCourtLitigatorFeesApiClient,
+        ApplicantBillingRepository applicantBillingRepository, ApplicantMapper applicantMapper) {
+        super(billingDataFeedLogService, crownCourtLitigatorFeesApiClient);
+      this.applicantBillingRepository = applicantBillingRepository;
+      this.applicantMapper = applicantMapper;
+    }
 
-        if (applicants.isEmpty()) {
-            return;
-        }
+    @Override
+    protected List getBillingDTOList() {
+        List<ApplicantBillingEntity> applicants = applicantBillingRepository.findAllApplicantsForBilling();
+        log.info("Extracted data for {} applicants", applicants.size());
 
-        List<Integer> ids = applicants.stream().map(ApplicantBillingDTO::getId).toList();
+        return applicants.stream().map(applicantMapper::mapEntityToDTO).toList();
+    }
 
-        resetApplicantBilling(
-            ResetApplicantBillingDTO.builder().userModified(userModified).ids(ids).build());
+    @Override
+    protected void resetBillingCCLFFlag(String userModified, List ids) {
```

Comment:
I know we have this naming elsewhere and this is my personal opinion, but I'm really not in love with the `CCLF` part of the method name. I think it should be something more like `resetBillingFlag` - we really don't need all these references to CCLF.